n3fit/src/evolven3fit/utils.py

```diff
@@ -0,0 +1,77 @@
+from scipy.interpolate import interp1d
+import numpy as np
+import math
+import pathlib
+import yaml
+
+
+class LhapdfLike:
```

Comment:
Maybe we should lift this into `ekomark`?

Reply:
You mean the "fake" LHAPDF class? Yes, but this is really specialized for this case, i.e. it has only one q2 (the fitting scale) and so the interpolation is one-dimensional (only on the x-grid). So i believe it is only useful for the fit. Anyway we can move it.

Reply:
Eventually (and if possible immediately) we should get rid of this class: in this case we don't want to reinterpolate at all, nor to approximate.

I.e. we should not use `apply_pdf`, but directly multiply the arrays as they are, user responsibility to use the same xgrid. But this is already understood and consistent: we decide to have a unique xgrid for a theory, so all the FkTables will be generated with a unique xgrid, and we'll use the same also for the EKO (in principle they could be different, but there is no reason). So, even the NN should be always evaluate on the same points, and the evolved result obtained just multiplying (at the end of the day, an EKO is only a wrapper around a bunch of numpy arrays).